x-pack/plugins/encrypted_saved_objects/server/create_migration.ts

```diff
@@ -62,24 +63,30 @@ export const getCreateMigration = (
       return encryptedDoc;
     }
 
-    const descriptor = {
-      id: encryptedDoc.id!,
-      type: encryptedDoc.type,
-      namespace: encryptedDoc.namespace,
-    };
+    // If a document has been converted right before this migration function is called, it will no longer have a `namespace` field, but it
+    // will have a `namespaces` field; in that case, the first/only element in that array should be used as the namespace in the descriptor
+    // during decryption.
+    const { convertToMultiNamespaceType } = inputType ?? {};
```

Comment:
The `getCreateMigration` function is called in the setup phase, and we don't have access to the saved objects type registry yet 😢 . Core only exposes the `core.savedObjects.registerType` function, which isn't very useful to us here.

That said, I think we can live without it. It would be nice to do extra validation like that but it's not completely necessary as long as our small number of consumers follow instructions. And we'll (I'll) be helping with the implementation of this!